pyAKI/kdigo.py

```diff
@@ -40,37 +71,61 @@ def __init__(
                 ),
                 DemographicsPreProcessor(stay_identifier=stay_identifier),
             ]
-
+        # apply preprocessors to the input data
         for preprocessor in preprocessors:
             data = preprocessor.process(data)
 
-        self._data = data
-        self._probes = probes
-        self._stay_identifier = stay_identifier
+        self._data: list[Dataset] = data
+        self._probes: list[Probe] = probes
+        self._stay_identifier: str = stay_identifier
 
     def process_stays(self) -> pd.DataFrame:
+        """
+        Process all stays in the input data.
+
+        This method processes all stays in the input data by applying the configured probes and preprocessors.
```

Comment:
Maybe ` This method processes all stays in the input data by applying the configured probes.` is more correct because the preprocessors are applied when initializing